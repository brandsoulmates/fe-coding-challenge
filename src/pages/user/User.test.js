import React from "react"
import { render } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import axios from "axios"

import { ProfileView } from "."

jest.mock('axios')

describe('UserPage component', () => {
  it('renders the test correctly', async () => {
    const mockData = {
      avatar_url: "https://avatars.githubusercontent.com/u/1?v=4",
      html_url: "https://github.com/mojombo",
      id: 1,
      login: "gituser",
      followers: 12,
      following: 32,
      public_gists: 273,
      public_repos: 46,
      twitter_username: "gituser",
      location: "united state",
      bio: "biography...",
      company: "google"
    }

    axios.get.mockResolvedValue({data: mockData})
    const result = await act( async () => render(<ProfileView username="gituser" />))
    expect(result).toMatchSnapshot()
  })
})
