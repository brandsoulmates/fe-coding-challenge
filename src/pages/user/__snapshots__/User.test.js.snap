// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage component renders the test correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="bg-sky-600"
      >
        <div
          class="mx-auto pt-5 max-w-5xl min-h-screen"
        >
          <div
            class="flex mt-6 py-8 pl-8 pr-16 gap-20 bg-white border-2 border-blue-500 rounded-2xl shadow-lg"
          >
            <div
              class="text-center"
            >
              <img
                alt="gituser"
                class="mx-auto w-48 h-48 rounded-full"
                src="https://avatars.githubusercontent.com/u/1?v=4"
              />
              <h1
                class="mt-4 text-xl font-medium"
              />
              <p
                class="mb-3"
              >
                gituser
              </p>
              <p>
                united state
              </p>
              <a
                class="text-blue-500 underline"
                href="https://github.com/mojombo"
              >
                https://github.com/mojombo
              </a>
            </div>
            <div
              class="flex-1 flex flex-col justify-between"
            >
              <div
                class="flex mt-12 w-full justify-between"
              >
                <div
                  class="text-center"
                >
                  <p
                    class="text-4xl font-light"
                  >
                    273
                  </p>
                  <h2
                    class="mt-2 text-xl font-medium"
                  >
                    Public Gists
                  </h2>
                </div>
                <div
                  class="text-center"
                >
                  <p
                    class="text-4xl font-light"
                  >
                    46
                  </p>
                  <h2
                    class="mt-2 text-xl font-medium"
                  >
                    Public Repos
                  </h2>
                </div>
                <div
                  class="text-center"
                >
                  <p
                    class="text-4xl font-light"
                  >
                    12
                  </p>
                  <h2
                    class="mt-2 text-xl font-medium"
                  >
                    Followers
                  </h2>
                </div>
                <div
                  class="text-center"
                >
                  <p
                    class="text-4xl font-light"
                  >
                    32
                  </p>
                  <h2
                    class="mt-2 text-xl font-medium"
                  >
                    Following
                  </h2>
                </div>
              </div>
              <div
                class="content"
              >
                <p>
                  <b>
                    Biography: 
                  </b>
                  biography...
                </p>
                <p>
                  <b>
                    Company: 
                  </b>
                  google
                </p>
                <p>
                  <b>
                    Twitter: 
                  </b>
                  <a
                    class="text-blue-500 underline"
                    href="https://twitter.com/gituser"
                  >
                    gituser
                  </a>
                </p>
              </div>
            </div>
          </div>
          <div
            class="flex mt-4 justify-center items-center gap-8"
          >
            <a
              class="text-white opacity-70 pointer-events-none"
              href="/user/"
            >
              &lt; Back
            </a>
            <div
              class="w-px h-5 bg-white"
            />
            <a
              class="text-white opacity-70 pointer-events-none"
              href="/user/"
            >
              Next &gt;
            </a>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="bg-sky-600"
    >
      <div
        class="mx-auto pt-5 max-w-5xl min-h-screen"
      >
        <div
          class="flex mt-6 py-8 pl-8 pr-16 gap-20 bg-white border-2 border-blue-500 rounded-2xl shadow-lg"
        >
          <div
            class="text-center"
          >
            <img
              alt="gituser"
              class="mx-auto w-48 h-48 rounded-full"
              src="https://avatars.githubusercontent.com/u/1?v=4"
            />
            <h1
              class="mt-4 text-xl font-medium"
            />
            <p
              class="mb-3"
            >
              gituser
            </p>
            <p>
              united state
            </p>
            <a
              class="text-blue-500 underline"
              href="https://github.com/mojombo"
            >
              https://github.com/mojombo
            </a>
          </div>
          <div
            class="flex-1 flex flex-col justify-between"
          >
            <div
              class="flex mt-12 w-full justify-between"
            >
              <div
                class="text-center"
              >
                <p
                  class="text-4xl font-light"
                >
                  273
                </p>
                <h2
                  class="mt-2 text-xl font-medium"
                >
                  Public Gists
                </h2>
              </div>
              <div
                class="text-center"
              >
                <p
                  class="text-4xl font-light"
                >
                  46
                </p>
                <h2
                  class="mt-2 text-xl font-medium"
                >
                  Public Repos
                </h2>
              </div>
              <div
                class="text-center"
              >
                <p
                  class="text-4xl font-light"
                >
                  12
                </p>
                <h2
                  class="mt-2 text-xl font-medium"
                >
                  Followers
                </h2>
              </div>
              <div
                class="text-center"
              >
                <p
                  class="text-4xl font-light"
                >
                  32
                </p>
                <h2
                  class="mt-2 text-xl font-medium"
                >
                  Following
                </h2>
              </div>
            </div>
            <div
              class="content"
            >
              <p>
                <b>
                  Biography: 
                </b>
                biography...
              </p>
              <p>
                <b>
                  Company: 
                </b>
                google
              </p>
              <p>
                <b>
                  Twitter: 
                </b>
                <a
                  class="text-blue-500 underline"
                  href="https://twitter.com/gituser"
                >
                  gituser
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          class="flex mt-4 justify-center items-center gap-8"
        >
          <a
            class="text-white opacity-70 pointer-events-none"
            href="/user/"
          >
            &lt; Back
          </a>
          <div
            class="w-px h-5 bg-white"
          />
          <a
            class="text-white opacity-70 pointer-events-none"
            href="/user/"
          >
            Next &gt;
          </a>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
